clang/lib/AST/ByteCode/Context.cpp

```diff
@@ -44,12 +44,12 @@ bool Context::isPotentialConstantExpr(State &Parent, const FunctionDecl *FD) {
   Compiler<ByteCodeEmitter>(*this, *P).compileFunc(
       FD, const_cast<Function *>(Func));
 
-  ++EvalID;
-  // And run it.
-  if (!Run(Parent, Func))
+  if (!Func->isValid())
     return false;
 
-  return Func->isValid();
+  ++EvalID;
+  // And run it.
```

Comment:
I wonder how useful this inline comments are, I feel like that might be condensed and added to the comment on the declaration of `isPotentialConstantExpr` something like 

"Check if the function is a potential constant expression by compiling the function and running it."

Reply:
I don't really care either way but this comment isn't new from this patch, just moved down a bit.

Reply:
This is a drive by nit but since you in this area already it feels like a worthwhile fix.